### PR TITLE
Core: Restore Docs `useParameter` using `DOCS_PREPARED`

### LINKED ISSUE #21798: [Bug]: useParameter() returns nothing for a docs page
### Describe the bug

I'm attempting to upgrade a storybook 6 project to storybook 7 and am running into an issue where the `storybook-dark-mode` doesn't work for docs pages. 

I've narrowed down the issue to `@storybook/api` `useParameter('darkMode')` returning an empty object inside the addon, but _only_ for rendering docs pages.

### To Reproduce

I don't have a reproduction at the moment, but a basic storybook v7 project with `storybook-dark-mode` added should exhibit the behavior. In chrome devtools I looked up the `manager.js` file from storybook-dark-mode and added a breakpoint where it uses the `useParameter` hook.

My guess is the parameters store isn't getting initialized for docs pages for some reason?

### System

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by tmeasday**
Closes #21798 

Telescoping on https://github.com/storybookjs/storybook/pull/22120

## What I did

Added a new event `DOCS_PREPARED` analogous to `STORY_PREPARED` that emits the *parameters* of the current docs entry.

 - That's the component's parameters for CSF docs (autodocs + `.stories.mdx`), and attached MDX docs.
 - That's the project's parameters for unattached MDX docs.

This is in contrast to https://github.com/storybookjs/storybook/pull/22071 which added more impactful events (`PROJECT_PREPARED` and `COMPONENT_PREPARED`) which we should think a little more about.

## How to test

1. Run a sandbox, check toolbar (in particular backgrounds) behaviour
	- Check attached MDX, unattached.
3. Check the same for composition!
4. See tests

## Checklist

- [ ] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [ ] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by probablyup** [comment]
Copied from the other thread:

I’d really like to avoid returning nothing during render to avoid FOUC inside the preview window. It would make the dark mode addon have imperfect UX since it couldn’t proactively set dark mode based on config in preview.js.

Really would just love parameters exported from preview to be the default, it would solve all problems I’m aware of.

**Comment by tmeasday** [comment]
> I’d really like to avoid returning nothing during render to avoid FOUC inside the preview window. It would make the dark mode addon have imperfect UX since it couldn’t proactively set dark mode based on config in preview.js.

My understand (based on playing with it and reading the code), is that is what the addon does already in 6.5, at least if you are using `storyStoreV7`.

> Really would just love parameters exported from preview to be the default, it would solve all problems I’m aware of. And why wouldn’t you? Isn’t anything in preview.js a global by default?

Yes but it can be overriden on a per-component or per story basis. So you could be browsing from one story to another that both disable an addon, and we could fallback to the project parameters which enable it, and weird things might happen.

**Comment by probablyup** [comment]
> My understanding (based on playing with it and reading the code), is that is what the addon does already in 6.5, at least if you are using storyStoreV7. Were you using ssv7 (code splitting/on demand store) in 6.5 @probablyup?

Have not been because CSF doesn’t support dynamic stories, have a big dark mode and variant story generation system I’d love to convert over to a CSF-equivalent when that someday becomes available. 

> Yes but it can be overridden on a per-component or per story basis. So you could be browsing from one story to another that both disable an addon, and we could briefly fallback to the project parameters which enable it, and weird things might happen.

This is less likely to happen imo.

**Comment by tmeasday** [comment]
> This is less likely to happen imo and addons can avoid this by listening to events...

OK, sure but making a change in a patch release is the issue here. I'm open to discussion on what behaviour we want in 7.1 and beyond.

**Comment by shilman** [comment]
There's a common pattern where addons (backgrounds, i18n, etc) store their state in SB globals, and then parameters set on the story level are used to update the value of the globals are updated by a decorator also provided by the addon.

This is admittedly ugly, but there's no other declarative way to update globals. Would love to find a good way to handle this in the future and it could put a whole swath of related issues--including this one--to bed. Easier said than done, of course!

**Review by shilman — APPROVED** [review]

File: code/lib/manager-api/src/tests/stories.test.ts [hunk 1]
```diff
@@ -1424,6 +1444,37 @@ describe('stories API', () => {
...
+      const fullAPI = Object.assign(new EventEmitter(), {
+        setStories: jest.fn(),
+        setOptions: jest.fn(),
+      });
+
+      const { api, init } = initStoriesAndSetState({ store, navigate, provider, fullAPI } as any);
+      Object.assign(fullAPI, api);
+
+      await init();
+      fullAPI.emit(STORY_PREPARED, {
```
[the remark this answers is not in the document]
Comment (tmeasday) [reply code/lib/manager-api/src/tests/stories.test.ts:1460]: Thanks @probablyup

**Pull request merged**